torchaudio/functional/functional.py

```diff
@@ -762,15 +771,22 @@ def mask_along_axis_iid(
     if axis not in [2, 3]:
         raise ValueError('Only Frequency and Time masking are supported')
 
+    if not 0.0 <= p <= 1.0:
+        raise ValueError(f"The value of p must be between 0.0 and 1.0 ({p} given).")
+
+    mask_param = _get_mask_param(mask_param, p, specgrams.shape[axis])
+    if mask_param < 1:
+        return specgrams
+
     device = specgrams.device
     dtype = specgrams.dtype
 
     value = torch.rand(specgrams.shape[:2], device=device, dtype=dtype) * mask_param
     min_value = torch.rand(specgrams.shape[:2], device=device, dtype=dtype) * (specgrams.size(axis) - value)
 
     # Create broadcastable mask
-    mask_start = min_value[..., None, None]
-    mask_end = (min_value + value)[..., None, None]
+    mask_start = min_value.long()[..., None, None]
+    mask_end = (min_value.long() + value.long())[..., None, None]
```

Comment:
made this fix to ensure consistency with `mask_along_axis`; without it, the resulting number of masked columns is not guaranteed to be strictly less than `mask_param`, and `test_mask_along_axis_iid` fails for certain random seeds